node/lib/util/diff_util.js

```diff
@@ -129,41 +129,38 @@ exports.getRepoStatus = co.wrap(function *(repo,
         options.flags = options.flags |
                         NodeGit.Diff.OPTION.RECURSE_UNTRACKED_DIRS;
     }
-    const indexDiff =
-               yield NodeGit.Diff.treeToIndex(repo, tree, null, options);
-    const indexStatus = readDiff(indexDiff);
     const index = yield repo.index();
     const workdirToIndexDiff =
                        yield NodeGit.Diff.indexToWorkdir(repo, index, options);
     const workdirToIndexStatus = readDiff(workdirToIndexDiff);
     if (!workdirToTree) {
+        const indexToTreeDiff =
+            yield NodeGit.Diff.treeToIndex(repo, tree, null, options);
+        const indexToTreeStatus = readDiff(indexToTreeDiff);
         return {
-            staged: indexStatus,
+            staged: indexToTreeStatus,
             workdir: workdirToIndexStatus,
         };
     }
-    const workdirDiff =
```

Comment:
Oh, I see. IIRC, GIt handles this with a temporary index which is just used during the commit.  But I guess this way works too.